lib/archethic/contracts/interpreter/library/common/http_impl.ex

```diff
@@ -0,0 +1,156 @@
+defmodule Archethic.Contracts.Interpreter.Library.Common.HttpImpl do
+  @moduledoc """
+  Http client for the Smart Contracts.
+  Implements AEIP-20.
+
+  Mint library is processless so in order to not mess with
+  other processes, we use it from inside a Task.
+  """
+
+  alias Archethic.Contracts.Interpreter.Library
+  alias Archethic.Contracts.Interpreter.Library.Common.Http
+  alias Archethic.TaskSupervisor
+
+  @behaviour Http
+  @threshold 256 * 1024
+
+  @impl Http
+  def fetch(uri) do
+    task =
+      Task.Supervisor.async_nolink(
+        TaskSupervisor,
+        fn -> do_fetch(uri) end
+      )
+
+    case Task.yield(task, 2_000) || Task.shutdown(task) do
```

Comment:
Maybe the timeout could be retrieved from config, so we would not need to wait 4 seconds in test to ensure timeout is reached